### ISSUE TITLE
Podman on windows looks in too many (wrong) places for configuration files

### ISSUE THREAD

**Issue by Tyler-IN** [podman-container-tools/podman#22875]
### Issue Description

Using [ProcMon](https://learn.microsoft.com/en-us/sysinternals/downloads/procmon) one can easily monitor process read attempts.

Some paths are not appropriately joined for the roaming app data or program data resulting in missing `\` in the 5.1.0-rc1, I will try the release build too.

Probably don't search in `C:\usr\share\containers` and `C:\etc\containers` on Windows builds

I think `<home>\.config\containers` should be fine.

### Steps to reproduce the issue

Steps to reproduce the issue
1. Launch ProcMan and configure filters to capture file activity with 'process name is podman.exe' and 'path ends with .conf' as filters.
2. Execute podman.exe
3. Observe captured results in ProcMon.

### Describe the results you received

The files being hunted are containers.conf and storage.conf...
* `C:\etc\containers\`
* `C:\usr\share\containers\`
* `C:\ProgramDatacontainers\`
* `C:\Users\<Username>\AppData\Roamingcontainers\`
* `C:\Users\<Username>\.config\containers\`

### Describe the results you expected

Not ... this.

### podman info output

```yaml
OS: windows/amd64
provider: wsl
version: 5.1.0-rc1

Cannot connect to Podman. (I have the machine shut down at the moment.)
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

No

**Comment by Tyler-IN** [comment]
Upgraded to released v5.1.0, problem persists.

It is also checking in `D:\etc` and `D:\usr\share\containers` for some reason.

**Comment by rhatdan** [comment]
Podman is searching for storage.conf in those directories. This should not be happening in Windows, but it is quite difficult to fix.

**Comment by ppoi** [comment]
5.0.3: `~\AppData\Roaming\containers`
5.1.0: `~\AppData\Roamingcontainers`

I think that 5.1.0's configuration folder path building process is wrong.

**Comment by Tyler-IN** [comment]
I investigated fixing it, fixed it in a local copy; mentioned the issue in https://github.com/containers/storage/issues/1943

Just needs to not use dumb paths on Windows and join paths with the right functions.

Might PR if I have time to ensure red tape is not an concern.

**Comment by unlimitedsola** [comment]
On Windows I think it should still check for `%USERPROFILE%\.config\containers\containers.conf` as a fallback. As of **5.5.2** it only checks `%APPDATA%\containers\containers.conf`. This confused me because there is no mention of `%APPDATA%` in <https://docs.podman.io/en/latest/markdown/podman-machine.1.html> and I was expecting it will check `%USERPROFILE%\.config\containers\containers.conf` from the wording of the man page.

Please let me know if I should file a new issue for this.

**Comment by boltomli** [comment]
yes it took me a while to copy the same conf to many folders before checking out the issue

**Comment by l0rd** [comment]
Thank you @Tyler-IN. This got fixed in main branch.

Testing with Podman 5.8.2:
```
podman.exe	10972	CreateFile	C:\etc\containers\storage.conf
podman.exe	10972	CreateFile	C:\usr\share\containers\storage.conf
podman.exe	10972	CreateFile	C:\Users\mario\.config\containers\storage.conf
podman.exe	10972	CreateFile	C:\ProgramData\containers\containers.conf
podman.exe	10972	CreateFile	C:\Users\mario\AppData\Roaming\containers\containers.conf
podman.exe	10972	ReadFile	C:\Users\mario\AppData\Roaming\containers\containers.conf
podman.exe	10972	ReadFile	C:\Users\mario\AppData\Roaming\containers\containers.conf
podman.exe	10972	CloseFile	C:\Users\mario\AppData\Roaming\containers\containers.conf
podman.exe	10972	CreateFile	C:\Users\mario\AppData\Roaming\containers\containers.conf.d\99-podman-machine-provider.conf
podman.exe	10972	ReadFile	C:\Users\mario\AppData\Roaming\containers\containers.conf.d\99-podman-machine-provider.conf
podman.exe	10972	ReadFile	C:\Users\mario\AppData\Roaming\containers\containers.conf.d\99-podman-machine-provider.conf
podman.exe	10972	CloseFile	C:\Users\mario\AppData\Roaming\containers\containers.conf.d\99-podman-machine-provider.conf
```

Testing with Podman 6.0.0-dev (current main branch):

```log
podman.exe	6136	CreateFile	C:\Users\mario\AppData\Roaming\containers\storage.conf
podman.exe	6136	CreateFile	C:\ProgramData\containers\storage.conf
podman.exe	6136	CreateFile	C:\Users\mario\AppData\Roaming\containers\containers.conf
podman.exe	6136	ReadFile	C:\Users\mario\AppData\Roaming\containers\containers.conf
podman.exe	6136	ReadFile	C:\Users\mario\AppData\Roaming\containers\containers.conf
podman.exe	6136	CloseFile	C:\Users\mario\AppData\Roaming\containers\containers.conf
podman.exe	6136	CreateFile	C:\Users\mario\AppData\Roaming\containers\containers.conf.d\99-podman-machine-provider.conf
podman.exe	6136	ReadFile	C:\Users\mario\AppData\Roaming\containers\containers.conf.d\99-podman-machine-provider.conf
podman.exe	6136	ReadFile	C:\Users\mario\AppData\Roaming\containers\containers.conf.d\99-podman-machine-provider.conf
podman.exe	6136	CloseFile	C:\Users\mario\AppData\Roaming\containers\containers.conf.d\99-podman-machine-provider.conf
```

The fix should be in one of [these commits (first page)](https://github.com/podman-container-tools/container-libs/commits/podman-6.0/common/pkg/config), probably a side-effect of https://github.com/podman-container-tools/container-libs/pull/615.
